src/orion/core/io/experiment_builder.py

```diff
@@ -200,34 +208,76 @@ def _instantiate_space(config):
     return SpaceBuilder().build(config)
 
 
-def _instantiate_algo(space, max_trials, config=None, ignore_unavailable=False):
+def _instantiate_knowledge_base(kb_config: dict[str, Any]) -> KnowledgeBase:
+    """Instantiate the Knowledge base from its configuration."""
+    if len(kb_config) != 1:
```

Comment:
I think a test for these different ways of defining the DB config should be relatively simple to test.

Reply:
Should be fixed by 34965b402c81ab37aad27c83e034780bc2e0f162

Reply:
Looks like there are multiple possible ways to incorrectly define the KB (3 different raises) but only 1 is covered.

Reply:
Fixed in 3f35bbaf092beadd874bf86cf3d212af8d48a79d